lib/providers/quoters/RoutingApiQuoter.ts

```diff
@@ -73,7 +120,7 @@ export class RoutingApiQuoter implements Quoter {
         tokenInChainId: request.info.tokenInChainId,
         tokenOutAddress: mapNative(request.info.tokenOut, request.info.tokenInChainId),
         tokenOutChainId: request.info.tokenOutChainId,
-        amount: request.info.amount.toString(),
+        amount: portionAdjustedRequestAmount,
```

Comment:
I think this is wrong actually. Even for exact out case, you should request for whatever clients requeste, which is ultimately from the swappers. 

By adding the portion amount into the requested amount for exact out, you might derive a quote route that might differ a bit from the original route. Also the inputAmount and outputAmount from the returned routes arrays will get inflated by the portion amount. When clients actually composed the calldata and submitted to universal router, swappers were left room for MEV bot exploitation.

We should be able to just keep `request.info.amount.toString()`, and then the entire implementation in `buildRequest` can also be simplified. This method reads like a convenience method with no outbound call, but it does due to portion. For simplicity, I think you can just pass in the optional `GetPortionResponse` object, and compute portionAmount there for exact out.